user profile redesign

krawczykkatarzyna opened the pull request:
Note: 
- breadcrumbs will be added in this pull request - https://github.com/JoinColony/colonyCDapp/pull/1440
- navigation on menu sidebar will be added later  

<img width="1175" alt="Screen Shot 2023-12-01 at 18 16 30" src="https://github.com/JoinColony/colonyCDapp/assets/119587095/06b2533e-a160-4943-aec0-c3d73ceaf51a">

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/JoinColony/colonyCDapp?pullRequest=1420) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/JoinColony/colonyCDapp?pullRequest=1420) before we can accept your contribution.<br/><hr/>**krawczykkatarzyna** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/JoinColony/colonyCDapp?pullRequest=1420) it.</sub>

melyndav reviewed (changes requested):
Hey @krawczykkatarzyna nice start on this.

**_Important note: This PR needs to be rebased but please ensure you do not override changes to this area that's been made for launch. We have hidden some content in this section for launch so please pay attention to this, thank you!._**

A few updates required.

1. Padding between the tabs and content block should be 24px across all tabs.

![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/c295f60c-3254-473a-95e1-161a73812d7c)

2. At 1440px breakpoint the content should sit right up on the left side. Currently (and it seems to be across all pages) there is extra padding here marked in red. 

<img width="61" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/11d629af-7615-4328-93ec-5ac9e6b1103a">

thomassdk commented:
Hi @krawczykkatarzyna thanks for the PR and sorry for the delay in getting a review on it can you rebase this ontop of master and ill have a look at it then

krawczykkatarzyna commented:
@thomassdk ready to check

melyndav reviewed (dismissed):
Hi @krawczykkatarzyna,

Thanks for the extra updates.

I'm still not seeing the breadcrumbs above the page title. When I load the account page directly I don't see the "Colony App" pagination. If I go to the home page I see the pagination, if I go back to the account page it is there. If I reload the page again though, then it is not there anymore.

![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/c356bb0f-8396-4043-9c5f-22d22f52e879)

![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/96d6167c-6832-47de-8fd8-e1ca3fdc0285)

krawczykkatarzyna commented:
@melyndav I wasn't responsible for creating breadcrumbs functionality, but I've add extra breadcrumbs for account page.

krawczykkatarzyna commented:
@bassgeta after rebase your approval disappeared, can you add approve again?

ArmandoGraterol commented on src/components/frame/v5/pages/UserProfilePage/partials/RowItem/RowItem.tsx:
```diff
@@ -0,0 +1,149 @@
+    );
+  }
+
+  if ('headerProps' in props) {
+    return (
+      <>
+        {headerProps && (
+          <div className="flex w-full">
+            <h5 className="heading-5">{headerProps?.title}</h5>
+          </div>
+        )}
+      </>
+    );
+  }
```
This snippet seems to be included already above so this can be removed

ArmandoGraterol reviewed (approved):
Looks good to me 👍

arrenv reviewed (approved):
All looks good, thank you!

jakubcolony commented on src/components/frame/v5/pages/UserProfilePage/partials/UserAccountForm/hooks.tsx:
```diff
@@ -0,0 +1,207 @@
+    }
+  };
+
+  /*
+   * Imperatively update the hasDisplayNameChanged form field, in order to be able to conditionally
+   * validate the displayName field. We don't want it to be validated if it hasn't changed.
+   */
+  const hasDisplayNameChanged = getValues('hasDisplayNameChanged');
+
+  if (displayNameDirty && !hasDisplayNameChanged) {
+    setValue('hasDisplayNameChanged', true, {
+      shouldValidate: false,
+    });
+  } else if (!displayNameDirty && hasDisplayNameChanged) {
+    setValue('hasDisplayNameChanged', false, {
+      shouldValidate: false,
+    });
+  }
```
Would it be possible to put the original display name in hook-form state (either as another field or preferably a value in the context provided to yup) so that the validation can derive `hasDisplayNameChanged` itself? This way the validation could be kept all in one place.

krawczykkatarzyna replied:
@arrenv I haven't changed anything in Tab component, just moved styles from module.css file 
@jakubcolony these are not my changes. I just moved these changes to the hook file.

krawczykkatarzyna replied:
@jakubcolony this is not may changes, I just moved this hook to different file. I suggest refactor it in separate pull request.

krawczykkatarzyna commented:
fixed reported issues

@arrenv I haven't changed anything in Tab component, just moved styles from module.css file
@jakubcolony these are not my changes. I just moved these changes to the hook file.

arrenv reviewed (dismissed):
I took this over to rebase and fixed an issue with the email form cancel and alignment. Approving based on everything else being addressed.

melyndav reviewed (dismissed):
Thanks for your work on these @krawczykkatarzyna & @arrenv. This is approved. :)

jakubcolony commented on src/components/frame/v5/pages/UserProfilePage/UserProfilePage.tsx:
```diff
@@ -1,59 +1,88 @@
+  const resolvedEditProfilePath = useResolvedPath(USER_EDIT_PROFILE_ROUTE);
+  const resolvedPreferencesPath = useResolvedPath(USER_PREFERENCES_ROUTE);
+  const resolvedAdvancedPath = useResolvedPath(USER_ADVANCED_ROUTE);
+  const [activeTab, setActiveTab] = useState<TabId>(TabId.Tab0);
```
Could you please use descriptive names for the tabs (enum members)? They seem to be Profile, Preferences, Advanced.

jakubcolony commented on src/components/frame/v5/pages/UserProfilePage/consts.ts:
```diff
@@ -1,3 +1,11 @@
+  USER_PREFERENCES_ROUTE,
+} from '~routes/routeConstants';
+
+export const tabRoutes = [
+  USER_EDIT_PROFILE_ROUTE,
+  USER_PREFERENCES_ROUTE,
+  USER_ADVANCED_ROUTE,
+];
```
This relies on the order of things matching the arbitrary tab indexes and is very easy to break, could you change it into a map between `TabId` and a corresponding route?

jakubcolony reviewed (changes requested):
Nice work, I was just wondering if you could improve the tabs logic a bit further.

jakubcolony reviewed (approved):
thanks for making the changes, LGTM 👍
